### PR TITLE
feat: Allow user setting of feature flags

### PR DESCRIPTION
**Pull request by ashtonG**
## Description

This allows users to set feature flags for themselves in the ui. Feature flags are stored in the settings table as a direct child of the `global-features` path, as either a boolean indicating whether the flag is on or off, or a null indicating that the value is unset.

## Test Plan

- [ ] visit an experiment list page
- [ ] open the settings drawer
- [ ] the experimental section should be at the bottom of the settings drawer
- [ ] when the "new experiment list" setting is set to on, the experiment list should show the new experiment list
- [ ] when the "new experiment list" setting is set to off, the experiment list should show the old experiment list
- [ ] when the "new experiment list" setting is set to unset, the experiment list should show the initial experiment list version.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
WEB-1360

### CODE REVIEW

File: webui/react/src/hooks/useFeature.ts [hunk 1]
```diff
@@ -20,19 +37,37 @@ interface FeatureHook {
...
+  let isOn: boolean = FeatureDefault[feature];
 
   // Read from config settings
   featureSwitches.includes(feature) && (isOn = true);
   featureSwitches.includes(`-${feature}`) && (isOn = false);
 
+  // Read from user settings
+  if (settings && feature in settings) {
+    const userValue = settings[feature];
+    // checks are split up bc typescript doesn't combine the typeguards properly
```
Comment (ashtonG) [inline webui/react/src/hooks/useFeature.ts:65]: quick typeescript playground link explaining what's happening in the code:
https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wGIFECCAVAVQCUsBlOAXjgG8AoORACwENYAuOAcgDlgB3OAGEWsVJwA09OKDAAbAJboA+gDcATBx784WcMCjyAtsGTwAMopgSpUMEoBG85ABMnAc01FgqCAFcoCMBwAAoQELJwAEJOrshu1gC+cMyoiCjoANy0MACeYEEAaswKzhjAzDD+QVQA1sA5EABmcLn5TXDY+MRktLQA9H0tjIpwIzCMQVDevrIw7i15QY1QEIajEAC0MKkbG6Pwhsx1qb4uwI1OwM7S4AoI8jDicKeN0JXIFcCyOdmLHeWVKakYAwOZxVCCFAXNyUGhSBgIEQwAD8HHsYVk5WQcAAPnBkDMInjTs5zpdnFkGAwZAplOpUXB0eEsbj8YTWSSycgrpSqbYHDF3AymZjmNi8QTZETnmcLtyKbQEr0ZG80mh4DBvDAMKcELCABR1HIcIolMoVKpPVAgsFuVAcSXS82A4DA0HuCFQ+RuACUlAAfHCGJj4CK0RiWVRGsVrb0GFJ5M19db3eC4AAydNwI2jbEp22pTNwfMegDaRoAunAAIQUKiOjNZkvg8v1Ku1qicuVXP10KlwAZwKbODitYBuXysZypFDFm1lytwZwoTjwMApVLjFa+NyMeGMjGw5t21s5CtSJUMRNwZPztNFnNOOepu29-dIdVwFTFXzVZ8F08K15K8k2-WRfxrOs2SlRsvx-IIOxlUlu2cN9+wYEVYTA3990vOAlRsEF-GxEVFSAA

Comment (ClaireNeveu) [reply webui/react/src/hooks/useFeature.ts:65]: Yeah, in order for it to work as one type Typescript would have to narrow the type of `settings` based on the latter checks which is very tricky.

File: webui/react/src/hooks/useFeature.ts [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+// add new feature switches here
+export const FEATURES = {
+  chart: 'New Charts',
+  explist_v2: 'New Experiment List',
+  rp_binding: 'Resource Pool Binding',
+} as const;
+export const FEATURE_SETTINGS_PATH = 'global-features';
 
-const FeatureDefault: { [K in ValidFeature]: boolean } = {
+const ioValidFeature = keyof(FEATURES);
```
Comment (ClaireNeveu) [inline webui/react/src/hooks/useFeature.ts:17]: Idiomatic approach is to name the io-ts Type the same as the typescript Type.

File: webui/react/src/hooks/useFeature.ts [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
 
 import determinedStore, { DeterminedInfo } from 'stores/determinedInfo';
+import userSettings from 'stores/userSettings';
+import { Loadable } from 'utils/loadable';
 
-// Add new feature switches below using `|`
-export type ValidFeature = 'dashboard' | 'explist_v2' | 'chart' | 'rp_binding';
+// add new feature switches here
+export const FEATURES = {
+  chart: 'New Charts',
```
Comment (ClaireNeveu) [inline webui/react/src/hooks/useFeature.ts:11]: 'Enable improved learning curve charts for experiment visualizations'
'Enable improved experiment listing, filtering, and comparison'
'Allow resource pools to be assigned to workspaces'

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]
The section is displaying the tooltips as the Feature names.
Screen Shot 2023-07-24 at 12 08 30

**Comment by ashtonG** [comment]
@ClaireNeveu are we not allowing the user to unset the features without going through the advanced ui?

**Comment by ClaireNeveu** [comment]
@ashtonG Alex decided that "use the cluster default" isn't a meaningful user need; people will just want it to be on or off. If they don't have a setting for it, just display the cluster state.

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]
Hate to be annoying but: can we give the feature switches a human readable name too?

Also it seems like with this change feature switches are just always on?

**Review by ClaireNeveu — APPROVED** [review]
After adding human readable names this looks good to me.

**Pull request merged**